portal-api/src/v1/users/repository.js

```diff
@@ -2,14 +2,24 @@ const { ObjectId } = require('mongodb');
 const db = require('../../drivers/db-client');
 const { transformDatabaseUser } = require('./transform-database-user');
 const { InvalidUserIdError, InvalidUsernameError, UserNotFoundError } = require('../errors');
+const { SIGN_IN_LINK_DURATION } = require('../../constants');
 
 class UserRepository {
   async saveSignInTokenForUser({ userId, signInTokenSalt, signInTokenHash }) {
     const saltHex = signInTokenSalt.toString('hex');
     const hashHex = signInTokenHash.toString('hex');
+    const expiry = new Date().getTime() + SIGN_IN_LINK_DURATION.MILLISECONDS;
 
     const userCollection = await db.getCollection('users');
-    return userCollection.updateOne({ _id: { $eq: ObjectId(userId) } }, { $set: { signInToken: { hashHex, saltHex } } });
+    return userCollection.updateOne({ _id: { $eq: ObjectId(userId) } }, { $set: { signInToken: { hashHex, saltHex, expiry } } });
+  }
+
+  async deleteSignInTokenForUser(userId) {
+    const userCollection = await db.getCollection('users');
+    return userCollection.updateOne(
+      { _id: { $eq: ObjectId(userId) } },
+      { $unset: { signInCode: '' } }
```

Comment:
```suggestion
      { $unset: { signInToken: '' } }
```

I think the field is called `signInToken`, not `signInCode`

Reply:
This has been changed.